File: windows/Claudario.Windows/Mascot/MascotPalette.cs
using SkiaSharp;

namespace Claudario.Windows.Mascot;

public record MascotColor(SKColor Body, SKColor Foot);

public static class MascotPalette
{
    // Matches macOS MascotPalette.swift color order exactly.
    // Index 0 is the default (Anthropic Orange).
    public static readonly MascotColor[] Colors =
    [
        new(SKColor.FromHsl(25,  90, 58), SKColor.FromHsl(25,  80, 42)),  // Anthropic Orange
        new(SKColor.FromHsl(204, 75, 55), SKColor.FromHsl(204, 65, 38)),  // Sky Blue
        new(SKColor.FromHsl(152, 55, 52), SKColor.FromHsl(152, 50, 36)),  // Mint
        new(SKColor.FromHsl(270, 55, 65), SKColor.FromHsl(270, 45, 48)),  // Lavender
        new(SKColor.FromHsl(6,   72, 60), SKColor.FromHsl(6,   62, 44)),  // Coral
        new(SKColor.FromHsl(42,  85, 55), SKColor.FromHsl(42,  75, 38)),  // Mustard
        new(SKColor.FromHsl(340, 65, 65), SKColor.FromHsl(340, 55, 48)),  // Rose
        new(SKColor.FromHsl(178, 58, 48), SKColor.FromHsl(178, 50, 32)),  // Teal
        new(SKColor.FromHsl(25,  40, 40), SKColor.FromHsl(25,  35, 28)),  // Cocoa
        new(SKColor.FromHsl(220, 18, 52), SKColor.FromHsl(220, 15, 36)),  // Slate
    ];
}

File: macos/Sources/Claudario/Mascot/MascotPalette.swift
import AppKit

struct MascotColor {
    let name: String
    let body: NSColor
    let foot: NSColor
}

enum MascotPalette {
    static let colors: [MascotColor] = [
        MascotColor(
            name: "Anthropic Orange",
            body: NSColor(calibratedRed: 0.95, green: 0.55, blue: 0.25, alpha: 1.0),
            foot: NSColor(calibratedRed: 0.45, green: 0.22, blue: 0.08, alpha: 1.0)),
        MascotColor(
            name: "Sky Blue",
            body: NSColor(calibratedRed: 0.40, green: 0.65, blue: 0.90, alpha: 1.0),
            foot: NSColor(calibratedRed: 0.18, green: 0.32, blue: 0.50, alpha: 1.0)),
        MascotColor(
            name: "Mint",
            body: NSColor(calibratedRed: 0.45, green: 0.80, blue: 0.55, alpha: 1.0),
            foot: NSColor(calibratedRed: 0.20, green: 0.42, blue: 0.25, alpha: 1.0)),
        MascotColor(
            name: "Lavender",
            body: NSColor(calibratedRed: 0.65, green: 0.55, blue: 0.85, alpha: 1.0),
            foot: NSColor(calibratedRed: 0.30, green: 0.22, blue: 0.45, alpha: 1.0)),
        MascotColor(
            name: "Coral",
            body: NSColor(calibratedRed: 0.95, green: 0.45, blue: 0.45, alpha: 1.0),
            foot: NSColor(calibratedRed: 0.50, green: 0.18, blue: 0.18, alpha: 1.0)),
        MascotColor(
            name: "Mustard",
            body: NSColor(calibratedRed: 0.90, green: 0.75, blue: 0.30, alpha: 1.0),
            foot: NSColor(calibratedRed: 0.45, green: 0.35, blue: 0.10, alpha: 1.0)),
        MascotColor(
            name: "Rose",
            body: NSColor(calibratedRed: 0.95, green: 0.55, blue: 0.70, alpha: 1.0),
            foot: NSColor(calibratedRed: 0.50, green: 0.22, blue: 0.32, alpha: 1.0)),
        MascotColor(
            name: "Teal",
            body: NSColor(calibratedRed: 0.30, green: 0.70, blue: 0.70, alpha: 1.0),
            foot: NSColor(calibratedRed: 0.10, green: 0.35, blue: 0.35, alpha: 1.0)),
        MascotColor(
            name: "Cocoa",
            body: NSColor(calibratedRed: 0.65, green: 0.45, blue: 0.30, alpha: 1.0),
            foot: NSColor(calibratedRed: 0.30, green: 0.18, blue: 0.10, alpha: 1.0)),
        MascotColor(
            name: "Slate",
            body: NSColor(calibratedRed: 0.55, green: 0.60, blue: 0.65, alpha: 1.0),
            foot: NSColor(calibratedRed: 0.25, green: 0.30, blue: 0.35, alpha: 1.0)),
    ]
}
